src/test/java/com/karankumar/bookproject/backend/entity/BookTest.java

```diff
@@ -0,0 +1,84 @@
+package com.karankumar.bookproject.backend.entity;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.service.BookService;
+import com.karankumar.bookproject.backend.service.PredefinedShelfService;
+import com.karankumar.bookproject.backend.service.TagService;
+import com.karankumar.bookproject.backend.utils.PredefinedShelfUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Bo Jacobs
+ */
+
+@IntegrationTest
+public class BookTest {
+
+    private static BookService bookService;
+    private static TagService tagService;
+
+    private static Book testBook;
+    private static Tag testTag;
+
+    @BeforeAll
+    public static void setup(@Autowired PredefinedShelfService predefinedShelfService,
+            @Autowired BookService bookService,
+            @Autowired TagService tagService) {
+
+        BookTest.bookService = bookService;
+        BookTest.tagService = tagService;
+
+        resetBookService();
+        resetTagService();
+
+        PredefinedShelf toRead = new PredefinedShelfUtils(predefinedShelfService).findToReadShelf();
+
+        testTag = createTag("test tag");
+        BookTest.tagService.save(testTag);
+        testBook = createBook("test Book", toRead);
+        BookTest.bookService.save(testBook);
+    }
+
+    private static void resetTagService() {
```

Comment:
Could you please save the tag (via the `tagService` in this method)? The method name may need to be renamed to better reflect this